sdk/storage/faq.md

```diff
@@ -0,0 +1,374 @@
+// The return value is URL-encoded
+auto url = containerClient.Geturl();
+// blobName should be UTF-8 encoded
+auto blobClient = containerClient.GetBlobClinet(blobName);
+
+for (auto page = blobContainerClient.ListBlobs(); page.HasPage(); page.MoveToNextPage()) {
+  for (auto& blob : page.Blobs) {
+    // blob.Name is UTF-8 encoded
+    std::cout << blob.Name << std::endl;
+  }
+}
+```
+
+## How can I check if a blob exists? Is there a function `Exists()` like in legacy SDK?
+
+There's no equivalent in the new C++ SDK. This function is usually just a convenient method on top of getting blob properties. You could easily write one like below.
+
+```C++
+bool BlobExists(const Azure::Storage::Blobs::BlobClient& client) {
+  try {
+    client.GetProperties();
+    return true;
+  }
+  catch (Azure::Storage::StorageException& e) {
+    if (e.ReasonPhrase == "The specified blob does not exist.") {
+      return false;
+    }
+    throw;
+  }
+}
+```
+
+A common use scenario is checking if a blob exists before writing to it.
+It is fine if you write from a single thread.
+However, it is problematic if the destination might be modified by multiple threads or processes. A subsequent write operation may get overwritten by the previous one.
+In this case, it's more recommended to:
+
+1. Use `CreateIfNotExists()`, `DeleteIfExists()` functions whenever possible. These functions internally use access conditions and can help us catch unexpected exceptions caused by resending PUT/DELETE requests on network errors.
+1. Use access conditions for other operations. Code below only sends one HTTP request, check-and-write is performed atomically. It only succeeds if the blob doesn't exist.
+   ```C++
+   UploadBlockBlobOptions options;
+   options.AccessConditions.IfNoneMatch = Azure::ETag::Any();
+   blobClient.Upload(stream, options);
+   ```
+1. Use lease for more complex scenarios that invole multiple operations on the same resource.
+
+## What is the difference between `BlockBlobClient::Upload()` and `BlockBlobClient::UploadFrom()`? When should I use one vs the other?
+
+`BlockBlobClient::Upload()` takes a stream as parameter and uploads the stream as a block blob with exact one HTTP request.
+The blob created with this method doesn't have any blocks, which means functions like `StageBlock()`, `CommitBlockList()` or `GetBlockList()` don't apply here.
+You cannot append blocks to such blobs (You'll have to overwrite it with a new one).
+You want to use this one if you need precise control over SDK behavior at HTTP level.
+
+`BlockBlobClient::UploadFrom` takes a memory buffer or file name as parameter, splits the data in the buffer or file into smaller chunks intelligently or according to some parameters if provided,
+then upload the chunks with multiple threads.
+This one suits in most cases. You can expect higher throughput because the chunks are transferred concurrently. It's especially recommended if you need to transfer large blobs efficiently.
```

Comment:
```suggestion
This one is suitable in most cases. You can expect higher throughput because the chunks are transferred concurrently. It's especially recommended if you need to transfer large blobs efficiently.
```